FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -18,7 +18,56 @@ public final class RemoteFeedLoader: FeedLoader {
 		self.client = client
 	}
 
+	private static let successCode = 200
+
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .success((let data, let response)):
+				if response.statusCode == RemoteFeedLoader.successCode, let items = RemoteFeedImageMapper.map(data) {
+					completion(.success(items))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
+			case .failure(_):
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+}
+
+private struct RemoteFeedImageMapper {
+	private struct Items: Decodable {
+		var items: [RemoteFeedImage]
+	}
+
+	static func map(_ data: Data) -> [FeedImage]? {
+		let remoteItems = try? JSONDecoder().decode(Items.self, from: data)
+		return remoteItems?.items.toFeedImage()
+	}
+}
+
+private struct RemoteFeedImage: Decodable {
+	enum CodingKeys: String, CodingKey {
+		case id = "image_id"
+		case description = "image_desc"
+		case location = "image_loc"
+		case url = "image_url"
+	}
+
+	let id: UUID
+	let description: String?
+	let location: String?
+	let url: URL
```

Comment:
It's ok to use the same API property names in the model because it's modeling exactly the API response.

So the code can be shorter and clear. The CodingKeys is not needed - you can remove it:

```suggestion
	let image_id: UUID
	let image_desc: String?
	let image_loc: String?
	let image_url: URL
```